core/src/test/kotlin/io/specmatic/stub/ThreadSafeListOfStubsTest.kt

```diff
@@ -190,4 +192,79 @@ class ThreadSafeListOfStubsTest {
         }
     }
 
+    @Nested
+    inner class PartialStubPrioritization {
+        @Test
+        fun `getPartialBySpecificityAndGenerality should select highest specificity`() {
+            // Create mock partial matches with different specificity values
+            val lowSpecificityRequest = HttpRequest("POST", "/products", body = parsedJSONObject("""{"name": "(string)"}"""))
+            val highSpecificityRequest = HttpRequest("POST", "/products", body = parsedJSONObject("""{"name": "Laptop"}"""))
+            
+            val lowSpecificityStub = HttpStubData(
+                requestType = lowSpecificityRequest.toPattern(),
+                response = HttpResponse.ok(parsedJSONObject("{\"id\": 1}")),
+                responsePattern = HttpResponsePattern(HttpResponse.OK),
+                resolver = Resolver(),
+                originalRequest = lowSpecificityRequest,
+                partial = ScenarioStub(request = lowSpecificityRequest, response = HttpResponse.ok(""))
```

Comment:
What your putting in HttpStubData.response should go into HttpStubData.partial.response.

Just confirm this by looking at how partials are constructed in ScenarioStub.mockFromJSON.